### DISCUSSION TITLE
建议：导出屏蔽信息，以便分析总结

### DISCUSSION THREAD

**Discussion by zhongfly** [Simple-Tracker/qBittorrent-ClientBlocker#48] [General]
我觉得可以考虑将屏蔽信息导出到文件，包含ip、peer id/名称、屏蔽原因这些信息。
细节方面我是这么设计的：每次屏蔽时，把信息追加到一个文本文件里去，一行一个，格式可以是ipfilter.dat，以便直接使用，ip为每行开头，然后#注释，后面是其他信息。
```
ip # peer id # peer 名称 # 原因
```
由于是追加的形式，就算程序重启，也可以一直积累数据，甚至可以每次启动时从文件里读取，设置为初始的屏蔽ip设置。

有这么一个文件也方便大家分享信息，或者是直接省事设置它作为qb的ip屏蔽数据。另外也方便总结情况，发现新的特征(我估计之后ip段会是最有用的特征)。

**Comment by IAALAI** [comment]
其实我一直有在这么干了。。。

这段是我从 qBittorrent-ClientBlocker 里提取 ipfilter.dat 的脚本，你可以参考参考 [gists](https://gist.github.com/IAALAI/fa92c95d5881660478d3ccc4228f3c24)
整个脚本很短，总共也没多少行。你可以自己扩展想要的功能，比如把封禁原因也加上去
但是对我来说意义不大，我自己用的时候是只封杀那一批恶意攻击的，其他的正常或者说接近正常的bt用户比如迅雷都是放行的
按照我的群友说的话：迅雷用户至少是有效流量，它下载了还会真的去看

**Comment by IAALAI** [comment]
对了，还有一个群友的小工具，给你推荐一下 https://github.com/vizv/ipfilter 可以实现多个 ipfilter.dat 合并,自动去重和组合IP范围

**Comment by Simple-Tracker** [comment]
程序在封禁时会打印日志, 这些日志默认会被按天记录到文件.  
因此, 通过适当的脚本解析日志, 就可实现提取信息.

另外, 我也在思考如何实现 IP 的扩大封禁, 如从一个 /32 变为 /24, 还有待进一步的测试.

**Comment by Simple-Tracker** [comment]
下一版本将支持在 Ban 及 Unban 时执行外部命令, 并可以向命令提供 PeerIP/PeerPort/TorrentInfoHash, 用户可以自行构建防火墙或统计及生成信息.

下下一版本可能会提供 IPBlockList.dat 的生成选项: 相比 IPFilter 的区别是, 它使用 IPCIDR 而非 IP-IP 的 range 范围.

Edit: 为了兼容其它 BT 客户端, 也会提供 IPFilter.dat 的兼容选项, 但其信息将不遵守 BanIPCIDR/BanIP6CIDR, 即不支持 CIDR.
